common/insightagent-boilerplate.py

```diff
@@ -0,0 +1,587 @@
+#!/usr/bin/env python
+import ConfigParser
+import collections
+import json
+import logging
+import os
+import re
+import socket
+import sys
+import time
+import pytz
+from optparse import OptionParser
+from itertools import islice
+from datetime import datetime
+import dateutil
+from dateutil.tz import tzlocal
+import urlparse
+import httplib
+import requests
+
+'''
+This script gathers data to send to Insightfinder
+'''
+
+
+def start_data_processing():
+    """ TODO: replace with your code.
+    Most work regarding sending to IF is abstracted away for you.
+    This function will get the data to send and prepare it for the API.
+    Programmer MUST specify track['mode'] here!
+    The general outline should be:
+    1. Declare track['mode'] (valid options given)
+    2. Gather data
+    3. Parse each entry
+    4. Call the appropriate handoff function 
+        metric_handoff()
+        log_handoff()
+        incident_handoff()
+    See zipkin for an example that uses os.fork to send both metric and log data.
+    """
+    track['mode'] = 'METRIC|METRICREPLAY|LOG|LOGREPLAY|INCIDENT|INCIDENTREPLAY'
+    
+    # you can optionally also set a timestamp format in track
+    track['timestamp_format'] = 'See datetime.strptime documentation'
+    
+
+def get_agent_config_vars():
+    """ Read and parse config.ini """
+    if os.path.exists(os.path.abspath(os.path.join(__file__, os.pardir, 'config.ini'))):
+        config_parser = ConfigParser.SafeConfigParser()
+        config_parser.read(os.path.abspath(os.path.join(__file__, os.pardir, 'config.ini')))
+        try:
+            ## TODO: fill out the fields to grab. examples given below
```

Comment:
What is to be pulled from a config could be passed into this function. If this file is to become an imported module, we need to make this standalone, not requiring any changes for specific data types

Reply:
see previous comment.